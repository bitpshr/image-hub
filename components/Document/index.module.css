@import "~lib/variables.css";

.root:nth-child(3n) {
	padding-left: calc((var(--grid-size) * 4) / 3);
}

.root:nth-child(3n - 1) {
	padding: 0 calc((var(--grid-size) * 4) / 6);
}

.root:nth-child(3n - 2) {
	padding-right: calc((var(--grid-size) * 4) / 3);
}

.small.root {
	padding: 0;
}

.card {
	border-radius: calc(var(--grid-size) / 4);
	border: 1px solid var(--input-border);
	margin-bottom: calc(var(--grid-size) * 2);
	padding: calc(var(--grid-size) * 2);
	width: 100%;
}

.filename {
	font-size: 20px;
	font-weight: bold;
	margin-bottom: calc(var(--grid-size) * 2.5);
	overflow: hidden;
	text-overflow: ellipsis;
	white-space: nowrap;
	width: 100%;
}
