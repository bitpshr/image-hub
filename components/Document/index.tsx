import { FunctionComponent, memo } from "react";

import Button from "@app/components/Button";
import Flex from "@app/components/Flex";
import { UploadDocument } from "@app/lib/interfaces";
import useMedia from "@app/lib/useMedia";
import { formatBytes } from "@app/lib/util";

import css from "./index.module.css";

export interface DocumentProps extends UploadDocument {
	onDelete: () => void;
}

const Document: FunctionComponent<DocumentProps> = memo((props) => {
	const size = useMedia();

	return (
		<Flex
			width={size.large ? "33.33%" : "100%"}
			className={`${css.root} ${!size.large ? css.small : ""}`}
		>
			<Flex
				alignItems="stretch"
				className={css.card}
				direction="column"
				grow={1}
			>
				<Flex>
					<div className={css.filename}>{props.filename}</div>
				</Flex>
				<Flex alignItems="center">
					<Flex grow={1}>{formatBytes(props.byteSize)}</Flex>
					<Button size="small" onClick={props.onDelete}>
						delete
					</Button>
				</Flex>
			</Flex>
		</Flex>
	);
});

export default Document;
