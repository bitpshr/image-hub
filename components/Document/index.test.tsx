import React from "react";
import renderer from "react-test-renderer";

import Document from "./";

test("default render", () => {
	const component = renderer.create(
		<Document
			byteSize={100000}
			date={1600202591670}
			filename="Doc1"
			onDelete={jest.fn()}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
