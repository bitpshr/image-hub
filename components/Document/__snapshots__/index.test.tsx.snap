// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default render 1`] = `
<div
  className="root small"
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <div
    className="card"
    style={
      Object {
        "alignItems": "stretch",
        "display": "flex",
        "flexDirection": "column",
        "flexGrow": 1,
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <div
        className="filename"
      >
        Doc1
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "flexDirection": "row",
            "flexGrow": 1,
          }
        }
      >
        100Kb
      </div>
      <button
        className="button small"
        onClick={[MockFunction]}
        size="small"
      >
        delete
      </button>
    </div>
  </div>
</div>
`;
